platforms/nuttx/src/px4/common/SerialImpl.cpp

```diff
@@ -74,6 +74,7 @@ bool SerialImpl::configure()
 	int speed;
 
 	switch (_baudrate) {
+	case 0: // special case, if baudrate is 0 it hangs entire system
```

Comment:
Shouldn't this be an error (return false) instead of silently falling through to a different baud rate?

Reply:
yes